tests/app/questionnaire/routing/test_when_rule_evaluator.py

```diff
@@ -0,0 +1,965 @@
+                    {Operator.DATE: ["now"]},
+                ]
+            },
+            True,
+        ),
+        (
+            {
+                Operator.LESS_THAN_OR_EQUAL: [
+                    {Operator.DATE: [now_as_yyyy_mm_dd, {"days": -1}]},
+                    {Operator.DATE: ["now", {"days": -1}]},
+                ]
+            },
+            True,
+        ),
+        (
+            {
+                Operator.GREATER_THAN: [
+                    {Operator.DATE: [now_as_yyyy_mm_dd]},
+                    {Operator.DATE: ["now", {"months": -1}]},
+                ]
+            },
+            True,
+        ),
+        (
+            {
+                Operator.GREATER_THAN_OR_EQUAL: [
+                    {Operator.DATE: [now_as_yyyy_mm_dd, {"months": -1}]},
+                    {Operator.DATE: ["now", {"months": -1}]},
+                ]
+            },
+            True,
+        ),
+        # Test inverse
+        (
+            {
+                Operator.EQUAL: [
+                    {Operator.DATE: [now_as_yyyy_mm_dd, {"days": -1}]},
+                    {Operator.DATE: [now_as_yyyy_mm_dd]},
+                ]
+            },
+            False,
+        ),
+        (
+            {
+                Operator.NOT_EQUAL: [
+                    {Operator.DATE: [now_as_yyyy_mm_dd]},
+                    {Operator.DATE: [now_as_yyyy_mm_dd]},
+                ]
+            },
+            False,
+        ),
+        (
+            {
+                Operator.LESS_THAN: [
+                    {Operator.DATE: [now_as_yyyy_mm_dd, {"days": 1}]},
+                    {Operator.DATE: ["now"]},
+                ]
+            },
+            False,
+        ),
+        (
+            {
+                Operator.GREATER_THAN: [
+                    {Operator.DATE: [now_as_yyyy_mm_dd, {"months": -1}]},
+                    {Operator.DATE: ["now"]},
+                ]
+            },
+            False,
+        ),
+        (
+            {
+                Operator.EQUAL: [
+                    {Operator.DATE: [now_as_yyyy_mm_dd]},
+                    {Operator.DATE: [None]},
+                ]
+            },
+            False,
+        ),
+    ],
+)
+def test_date_value(rule, expected_result):
+    when_rule_evaluator = get_when_rule_evaluator(
+        rule=rule,
+        answer_store=AnswerStore(
+            [
+                {
+                    "answer_id": "some-answer",
+                    "value": now_as_yyyy_mm_dd,
+                }
+            ]
+        ),
+        metadata={"some-metadata": now_as_yyyy_mm_dd},
+    )
+
+    assert when_rule_evaluator.evaluate() is expected_result
+
+
+@pytest.mark.parametrize(
+    "is_repeating_answer, list_item_id_for_answer, expected_result",
+    [
+        (True, None, False),
+        (True, "item-2", False),
+        (True, "item-1", True),
+        (False, None, True),
+        (False, "item-2", False),
+        (False, "item-1", False),
```

Comment:
I guess the confusion is from the way it is used. The idea is that it is just the `list_item_id_for_answer` used in the `AnswerStore`. So that when not in a repeat, the answer store might still have an answer but with list_item_id, and this ensures it doesn't fetch that and alternatively when answer is repeating, and answer store doesn't have an answer with matching list item id, it returns False. The `list_item_id_for_answer` is used for the Answer in the answer store and not the location itself, that might be where the confusion is. Happy to swap it around so it is the location that changes, if it helps.

Reply:
As discussed, this has been simplified and now is called `test_rule_access_answer_outside_of_repeating_section`. The other scenarios are already tested between the existing tests.

Reply:
We now have have some similar tests with inconsistent names:
```
test_rule_access_answer_outside_of_repeating_section
test_answer_source_with_answer_not_on_path_non_repeating_section
test_answer_source_with_answer_not_on_path_repeating_section
```

Maybe:
```
test_answer_source_outside_of_repeating_section
test_answer_source_not_on_path_non_repeating_section
test_answer_source_not_on_path_repeating_section
```

Reply:
Done. Since these also existed in value source resolver tests in terms of resolving, I have updated those also.